### PR TITLE
decoders/netflow: fix replacing current templates with incoming due to same pointer on [ ]Fields type.

### PR DESCRIPTION
**Pull request by kbatyr**

### CODE REVIEW

File: decoders/netflow/netflow.go [hunk 1]
```diff
@@ -355,15 +355,26 @@ func (ts *BasicTemplateSystem) AddTemplate(version uint16, obsDomainId uint32, t
 		ts.templates[version][obsDomainId] = make(map[uint16]interface{})
 	}
 	var templateId uint16
-	switch templateIdConv := template.(type) {
+	switch record := template.(type) {
 	case IPFIXOptionsTemplateRecord:
```
Comment (axlts) [inline decoders/netflow/netflow.go:359]: You're duplicating code in both cases. Merge them into one block:
```go
case IPFIXOptionsTemplateRecord, NFv9OptionsTemplateRecord:
```

Comment (axlts) [inline decoders/netflow/netflow.go:359]: You're duplicating code in both cases. Merge them into one block:
```go
case IPFIXOptionsTemplateRecord, NFv9OptionsTemplateRecord:
```

**Review by axlts — CHANGES_REQUESTED** [review]

File: decoders/netflow/netflow.go (hunk 1 above)
Comment (kbatyr) [reply decoders/netflow/netflow.go:359]: I tried, but the compiler does not allow to do this:
Screenshot from 2024-09-27 16-21-12

Comment (axlts) [reply decoders/netflow/netflow.go:359]: Surprisingly, it doesn't work with `type switch`. Sorry for having bothered you.

**Review by axlts — APPROVED** [review]

File: decoders/netflow/netflow.go (hunk 1 above)
Comment (axlts) [reply decoders/netflow/netflow.go:359]: Surprisingly, it doesn't work with a `type switch`. Sorry for having bothered you.

**Pull request merged**